### PR TITLE
ST_AsText : adding optional argument 'precision' 

### PR DESCRIPTION
**Pull request by marcu**
The optional argument 'precision' is used for specifying the number of significant digits in the output doubles

### CODE REVIEW

**Comment by mwtoews** [comment]
This is related to http://trac.osgeo.org/postgis/ticket/407 enhancement idea, which woud use `extra_float_digits` for all ST_As* fuctions. I wonder if a default should be obtained from that global.

File: postgis/lwgeom_ogc.c [hunk 1]
```diff
@@ -848,14 +848,20 @@ Datum LWGEOM_asText(PG_FUNCTION_ARGS)
...
 	POSTGIS_DEBUG(2, "Called.");
 
 	geom = PG_GETARG_GSERIALIZED_P(0);
 	lwgeom = lwgeom_from_gserialized(geom);
 
+	if (PG_NARGS() > 1)
+	{
+		dbl_dig_for_wkt = PG_GETARG_INT32(1);
+	}
+
```
Comment (mwtoews) [inline postgis/lwgeom_ogc.c]: Continuing on with the idea from https://trac.osgeo.org/postgis/ticket/407, a default precision could be placed in the "else" part here by retrieving the `extra_float_digits` global, and something linke `dbl_dig_for_wkt = DBL_DIG + extra_float_digits;` or `dbl_dig_for_wkt += extra_float_digits`

This is based on the [PostgreSQL docs](http://www.postgresql.org/docs/current/static/runtime-config-client.html)
> The parameter value [extra_float_digits] is added to the standard number of digits (`FLT_DIG` or `DBL_DIG` as appropriate).

**Comment by dbaston** [comment]
Ideally `ST_AsText` would behave similarly to other text-producing functions....so the current output of `POINT(1.1e+02 1.1)` in the doc example should be `111.11` to be consistent with functions like `ST_AsKML` and `ST_AsGeoJSON`.

**Comment by dbaston** [comment]
> I wonder if a default should be obtained from that global.

Whether we need/want to change the defaults is probably outside the scope of this improvement.

**Comment by mwtoews** [comment]
Right, setting defaults is not directly related to this enhancement, and should be ignored here.

**Comment by vesnikos** [comment]
At some cases, the scientific format might cause some issues. See 
http://gis.stackexchange.com/questions/190732/postgis-floating-point-representation-problem-with-st-astext

**Comment by robe2** [comment]
Are we done with discussion of this?  Is this ready for commit?

File: regress/out_geography_expected [hunk 2]
```diff
@@ -66,3 +66,5 @@ geojson_options_13|{"type":"LineString","crs":{"type":"name","properties":{"name
 geojson_options_14|{"type":"LineString","crs":{"type":"name","properties":{"name":"urn:ogc:def:crs:EPSG::4326"}},"coordinates":[[1,1],[2,2],[3,3],[4,4]]}
 geojson_options_15|{"type":"LineString","crs":{"type":"name","properties":{"name":"urn:ogc:def:crs:EPSG::4326"}},"bbox":[1,1,4,4],"coordinates":[[1,1],[2,2],[3,3],[4,4]]}
 geojson_options_16|{"type":"LineString","crs":{"type":"name","properties":{"name":"urn:ogc:def:crs:EPSG::4326"}},"bbox":[1,1,4,4],"coordinates":[[1,1],[2,2],[3,3],[4,4]]}
+text_precision_01|POINT(111.1111111 1.1111111)
+text_precision_02|POINT(1.1e+2 1.1)
```
Comment (Komzpa) [inline regress/out_geography_expected]: is 1.1e+2 a desired result in this case?

**Comment by Komzpa** [comment]
To get this merged we need:

- [ ] ticket in trac
- [ ] rebase on trunk
- [ ] change precision to mean "number of signs after the point" similar to ST_AsGeoJSON
- [ ] get rid of scientific format, test on values too large (100000000000) and too small (0.000000001)

@marcu do you need any help with that? :)

**Comment by Komzpa** [comment]
ticketed: https://trac.osgeo.org/postgis/ticket/3986

**Comment by Komzpa** [comment]
@marcu I did the switch from "number of significant digits" to "number of digits after comma" in https://github.com/postgis/postgis/pull/190. Are you available for a rebase of this PR on top of that one? :)

**Comment by Komzpa** [comment]
Let's go to #194 which has the branch rebased. (wow, exactly 100 pull requests later!)

**Pull request closed without merging**